core/arch/arm/mm/tee_pager.c

```diff
@@ -1012,10 +1035,16 @@ static void tee_pager_hide_pages(void)
 		} else
 			a = TEE_MATTR_HIDDEN_BLOCK;
 		area_set_entry(pmem->area, pmem->pgidx, pa, a);
+
+		va_tbl[va_cnt++] = area_idx2va(pmem->area, pmem->pgidx);
```

Comment:
Normally (except at boot and perhaps some corner case) we'll only hide one page at a time so this is a bit complicated.